codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionDecoder.java

```diff
@@ -51,6 +51,11 @@
     void frameListener(Http2FrameListener listener);
 
     /**
+     * Get the {@link Http2FrameListener} which will be notified when frames are decoded.
+     */
+    Http2FrameListener frameListener();
```

Comment:
If the decoder is provided to the builder, then the listener may already be set on the decoder, and not on the builder. We check if the decoder already has a listener before over writing it.

I guess we could just enforce that the builder always sets the frame listener...this user may then be responsible for pulling the listener out of and existing decoder...which would effectively set the listener to the same reference...let me do this for now.